src/utils/specs.py

```diff
@@ -34,67 +34,106 @@
 from pathlib import Path as path
 from textwrap import dedent
 
-gpu_list = {"gfx906", "gfx908", "gfx90a", "gfx900"}
-
-
 @dataclass
 class MachineSpecs:
     hostname: str
-    cpu: str
-    kernel: str
+    CPU: str
+    sbios: str
+    kernel_version: str
     ram: str
     distro: str
-    rocmversion: str
+    rocm_version: str
     GPU: str
+    arch: str
+    vbios: str
     L1: str
     L2: str
-    SCLK: str
     CU: str
     SIMD: str
     SE: str
     wave_size: str
-    workgroup_size: str
-    cur_SCLK: str
-    cur_MCLK: str
-    wave_occu: str
+    workgroup_max_size: str
+    max_sclk: str
+    cur_sclk: str
+    cur_mclk: str
+    max_waves_per_cu: str
+    L2Banks: str
+    LDSBanks: str
+    numSQC: str
+    hbmBW: str
+    compute_partition: str
+    memory_partition: str
 
     def __str__(self):
         return dedent(
             f"""\
         Host info:
             hostname:       {self.hostname}
-            cpu info:       {self.cpu}
+            CPU:            {self.CPU}
+            sbios:          {self.sbios}
             ram:            {self.ram}
             distro:         {self.distro}
-            kernel version: {self.kernel}
-            rocm version:   {self.rocmversion}
+            kernel_version: {self.kernel_version}
+            rocm_version:   {self.rocm_version}
         Device info:
             GPU:                {self.GPU}
-            L1:                 {self.L1}
-            L2:                 {self.L2}
-            Max SCLK:           {self.SCLK}MHz
-            Current SCLK:       {self.cur_SCLK}MHz
-            Current MCLK:       {self.cur_MCLK}MHz
+            arch:               {self.arch}
+            vbios:              {self.vbios}
+            L1:                 {self.L1} KB
+            L2:                 {self.L2} KB
+            max_sclk:           {self.max_sclk} MHz
+            cur_sclk:           {self.cur_sclk} MHz
+            cur_mclk:           {self.cur_mclk} MHz
             CU:                 {self.CU}
             SIMD:               {self.SIMD}
             SE:                 {self.SE}
-            Wave Size:          {self.wave_size}
-            Workgroup Max Size: {self.workgroup_size}
-            Max Wave Occupancy Per CU: {self.wave_occu}
+            wave_size:          {self.wave_size}
+            workgroup_max_size: {self.workgroup_max_size}
+            max_waves_per_cu:   {self.max_waves_per_cu}
+            L2Banks:            {self.L2Banks}
+            LDSBanks:           {self.LDSBanks}
+            numSQC:             {self.numSQC}
+            hbmBW:              {self.hbmBW} MB/s
+            compute_partition:  {self.compute_partition}
+            memory_partition:   {self.memory_partition}
         """
         )
 
 
 def gpuinfo():
-    rocminfo = run(["rocminfo"]).split("\n")
+    # Local var only for rocminfo searching
+    gpu_list = {"gfx906", "gfx908", "gfx90a", "gfx940", "gfx941", "gfx942"}
```

Comment:
This seems like a second authoritative list of SoC's that has to be maintained, whereas we previously said they are defined in: https://github.com/AMDResearch/omniperf/blob/131429087b1fe7ec6997c554c7e2f9069d8a057b/src/omniperf_base.py#L50-L57.

Reply:
Good catch. We'll import this from omniperf_base now.
https://github.com/AMDResearch/omniperf/blob/cbbd4fe75eeda296e5e4b083a5019a8a399e12e5/src/utils/specs.py#L107

https://github.com/AMDResearch/omniperf/blob/cbbd4fe75eeda296e5e4b083a5019a8a399e12e5/src/utils/specs.py#L132-L140